src/plugins/index_pattern_field_editor/public/components/preview/documents_nav_preview.tsx

```diff
@@ -45,40 +41,12 @@ export const DocumentsNavPreview = () => {
   // document ID as at that point there is no more reference to what's "next"
   const showNavButtons = isCustomId === false;
 
-  const onDocumentIdChange = useCallback((e: React.SyntheticEvent<HTMLInputElement>) => {
-    setIsCustomId(true);
-    const nextId = e.currentTarget.value;
-    setDocumentId(nextId);
-  }, []);
-
-  const loadDocFromCluster = useCallback(() => {
-    lastDocumentLoaded.current = null;
-    setIsCustomId(false);
-    loadFromCluster();
-  }, [loadFromCluster]);
-
-  useEffect(() => {
-    if (currentDocument && !isCustomId) {
-      setDocumentId(currentDocument._id);
-    }
-  }, [currentDocument, isCustomId]);
-
-  useDebounce(
```

Comment:
I have moved the logic to fetch the document inside the preview context file to have a single place to look at for the debounce and fetching. This also allows us to keep the state even if the nav is unmounted (for example when showing the empty prompt)